app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollStateTest.kt

```diff
@@ -184,60 +184,60 @@ class HorizontalSnapScrollStateTest {
             )
         ) {
             val scrollThresholdWidth = (columnWidth * SCROLL_THRESHOLD_FACTOR).toInt()
-            assertThat(calculateOnTouchColumnIndex(scrollX = scrollThresholdWidth + 1)).isEqualTo(0)
+            assertThat(calculateOnTouchColumnIndex(scrollX = -scrollThresholdWidth - 1)).isEqualTo(-1)
         }
     }
 
     @Test
-    fun `calculateOnTouchColumnIndex returns 0 as the minimum index in landscape mode to prevent scrolling off the screen to the left`() {
+    fun `calculateOnTouchColumnIndex returns -1 if the target index is lower than 0 in portrait mode to prevent scrolling off the screen to the left`() {
         with(
             createState().copy(
                 columnWidth = 993,
-                displayColumnCount = 4,
+                displayColumnCount = 1,
                 activeColumnIndex = 0,
-                roomsCount = 5
+                roomsCount = 10
             )
         ) {
-            assertThat(calculateOnTouchColumnIndex(scrollX = columnWidth * 2)).isEqualTo(0)
+            val scrollThresholdWidth = (columnWidth * SCROLL_THRESHOLD_FACTOR).toInt()
+            assertThat(calculateOnTouchColumnIndex(scrollX = scrollThresholdWidth + 1)).isEqualTo(-1)
         }
     }
 
     @Test
-    fun `calculateOnTouchColumnIndex returns the last possible index as the maximum value in portrait mode to prevent scrolling off the screen to the right`() {
+    fun `calculateOnTouchColumnIndex returns -1 if the target index is higher than the max allowed in landscape mode to prevent scrolling off the screen to the right`() {
         with(
             createState().copy(
                 columnWidth = 993,
-                displayColumnCount = 1,
-                activeColumnIndex = 4,
+                displayColumnCount = 4,
+                activeColumnIndex = 5,
                 roomsCount = 10
             )
         ) {
-            val scrollThresholdWidth = (columnWidth * SCROLL_THRESHOLD_FACTOR).toInt()
-            assertThat(calculateOnTouchColumnIndex(scrollX = -scrollThresholdWidth + 1)).isEqualTo(4)
+            assertThat(calculateOnTouchColumnIndex(scrollX = -columnWidth * 2)).isEqualTo(-1)
         }
     }
 
     @Test
-    fun `calculateOnTouchColumnIndex returns the last possible index as the maximum value in landscape mode to prevent scrolling off the screen to the right`() {
+    fun `calculateOnTouchColumnIndex returns -1 if the target index is lower than 0 in landscape mode to prevent scrolling off the screen to the left`() {
         with(
             createState().copy(
                 columnWidth = 993,
                 displayColumnCount = 4,
-                activeColumnIndex = 5,
+                activeColumnIndex = 1,
                 roomsCount = 10
             )
         ) {
-            assertThat(calculateOnTouchColumnIndex(scrollX = -columnWidth * 2)).isEqualTo(6)
+            assertThat(calculateOnTouchColumnIndex(scrollX = columnWidth * 2)).isEqualTo(-1)
```

Comment:
Why did you need to change the input values for these tests so dramatically? You could explain this in the commit message.

Reply:
That wasn't intentional, although `calculateOnTouchColumnIndex returns -1 if the target index is higher than the max allowed in portrait mode to prevent scrolling off the screen to the right` seems have been faulty as it would not produce the edge case that it should